yarn-project/noir-compiler/src/cli/codegen.ts

```diff
@@ -3,36 +3,48 @@ import path from 'path';
 
 import { generateContractArtifact } from '../contract-interface-gen/abi.js';
 import { generateTypescriptContractInterface } from '../contract-interface-gen/contractTypescript.js';
+import { generateNoirContractInterface } from '../contract-interface-gen/noir.js';
 
 /**
  *
  */
-export function generateTypescriptInterface(outputPath: string, fileOrDirPath: string, includeDebug = false) {
+export function generateCode(outputPath: string, fileOrDirPath: string, includeDebug = false, ts = false, nr = false) {
   const stats = statSync(fileOrDirPath);
 
   if (stats.isDirectory()) {
     const files = readdirSync(fileOrDirPath).filter(file => file.endsWith('.json') && !file.startsWith('debug_'));
     for (const file of files) {
       const fullPath = path.join(fileOrDirPath, file);
-      generateTypescriptInterfaceFromNoirAbi(outputPath, fullPath, includeDebug);
+      generateFromNoirAbi(outputPath, fullPath, includeDebug, ts, nr);
     }
   } else if (stats.isFile()) {
-    generateTypescriptInterfaceFromNoirAbi(outputPath, fileOrDirPath, includeDebug);
+    generateFromNoirAbi(outputPath, fileOrDirPath, includeDebug, ts, nr);
   }
 }
 
 /**
  *
  */
-function generateTypescriptInterfaceFromNoirAbi(outputPath: string, noirAbiPath: string, includeDebug: boolean) {
+function generateFromNoirAbi(outputPath: string, noirAbiPath: string, includeDebug: boolean, ts: boolean, nr: boolean) {
   const contract = JSON.parse(readFileSync(noirAbiPath, 'utf8'));
   const noirDebugPath = includeDebug ? getDebugFilePath(noirAbiPath) : undefined;
   const debug = noirDebugPath ? JSON.parse(readFileSync(noirDebugPath, 'utf8')) : undefined;
   const aztecAbi = generateContractArtifact({ contract, debug });
-  const tsWrapper = generateTypescriptContractInterface(aztecAbi, `./${aztecAbi.name}.json`);
+
   mkdirSync(outputPath, { recursive: true });
-  writeFileSync(`${outputPath}/${aztecAbi.name}.ts`, tsWrapper);
+
+  if (nr) {
+    const noirContract = generateNoirContractInterface(aztecAbi);
+    writeFileSync(`${outputPath}/${aztecAbi.name}.nr`, noirContract);
+    return;
```

Comment:
```suggestion
```

Unless we want to prevent running `--nr` and `--ts` simultaneously for some reason

Reply:
yeah, if specify --nr, you *dont* want the transformed json abi being output.
basically,
* no flag = abi (to be deprecated)
* --ts = abi + ts interface (to eventually become just ts interface)
* --nr = just nr interface

Reply:
I'm not sure I understand why you wouldn't want the abi being output if you also run --nr, but if we do, we should at least throw or warn if the user has set both --nr and --ts. Otherwise, the cli will just output the nr without any indication to the user.

Reply:
perhaps i should have left them as distinct commands actually, as the two languages are mutually exclusive.